src/CitizenProject/Voter/CommentCitizenProjectVoter.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace AppBundle\CitizenProject\Voter;
+
+use AppBundle\CitizenProject\CitizenProjectPermissions;
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\CitizenProject;
+
+class CommentCitizenProjectVoter extends AbstractCitizenProjectVoter
+{
+    protected function supports($attribute, $citizenProject)
+    {
+        return in_array($attribute, [CitizenProjectPermissions::COMMENT, CitizenProjectPermissions::SHOW_COMMENT], true)
+               && $citizenProject instanceof CitizenProject;
+    }
+
+    protected function doVoteOnAttribute(string $attribute, Adherent $adherent, CitizenProject $citizenProject): bool
```

Comment:
/!\ I'm not sure you can add type hint on an abstract method signature.

Reply:
The method signature is not changed. Check the base class ;-)

Reply:
Ah yes! Sorry!